### PR TITLE
Update the simple game server to move itself back to the Ready state after allocation

### PR DESCRIPTION
**Pull request by roberthbailey**
This allows us to test the "Reusing Allocated GameServers" integration pattern.

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature

> /kind hotfix

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by roberthbailey** [comment]
Flake in e2e-feature-gates:

```
--- FAIL: TestAutoscalerStressCreate (313.73s)
    framework.go:284: 
        	Error Trace:	framework.go:284
        	            				fleetautoscaler_test.go:355
        	            				fleetautoscaler_test.go:363
        	Error:      	Received unexpected error:
        	            	timed out waiting for the condition
        	Test:       	TestAutoscalerStressCreate
        	Messages:   	error waiting for fleet condition on fleet: simple-fleet-gt2xm
...
--- FAIL: TestFleetAutoscalerTLSWebhook (316.14s)
    framework.go:284: 
        	Error Trace:	framework.go:284
        	            				fleetautoscaler_test.go:709
        	Error:      	Received unexpected error:
        	            	timed out waiting for the condition
        	Test:       	TestFleetAutoscalerTLSWebhook
        	Messages:   	error waiting for fleet condition on fleet: simple-fleet-6jspb
FAIL
```

**Comment by roberthbailey** [comment]
Flake in e2e-stable:

```
--- FAIL: TestGameServerAllocationDuringMultipleAllocationClients (347.23s)
    framework.go:284: 
        	Error Trace:	framework.go:284
        	            				gameserverallocation_test.go:544
        	Error:      	Received unexpected error:
        	            	timed out waiting for the condition
        	Test:       	TestGameServerAllocationDuringMultipleAllocationClients
        	Messages:   	error waiting for fleet condition on fleet: preferred-9k64r
```

**Comment by roberthbailey** [comment]
Another flake in e2e-feature-gates:

```
--- FAIL: TestFleetAutoscalerTLSWebhook (313.26s)
    framework.go:284: 
        	Error Trace:	framework.go:284
        	            				fleetautoscaler_test.go:709
        	Error:      	Received unexpected error:
        	            	timed out waiting for the condition
        	Test:       	TestFleetAutoscalerTLSWebhook
        	Messages:   	error waiting for fleet condition on fleet: simple-fleet-c4bw7
FAIL
```

**Review by cindy52 — APPROVED** [review]

**Comment by roberthbailey** [comment]
I've been getting a weird error in the `make test` target:

```
/usr/share/dotnet/sdk/2.2.402/NuGet.targets(123,5): error : '1.20.0-0392579' is not a valid version string. [/go/src/agones.dev/agones/sdks/csharp/test/csharp-sdk-test.csproj]
/usr/share/dotnet/sdk/2.2.402/NuGet.targets(123,5): error : Parameter name: value [/go/src/agones.dev/agones/sdks/csharp/test/csharp-sdk-test.csproj]
make[1]: *** [includes/sdk.mk:88: run-sdk-command] Error 1
make: *** [includes/sdk.mk:84: run-sdk-command-csharp] Error 2
```

[Microsoft's documentation](https://docs.microsoft.com/en-us/nuget/concepts/package-versioning) says that the suffix is an optional string and that it should follow the semantic versioning convention. According to https://semver.org/spec/v1.0.0.html#spec-item-4

> A pre-release version number MAY be denoted by appending an arbitrary string immediately following the patch version and a dash. The string MUST be comprised of only alphanumerics plus dash [0-9A-Za-z-].

The suffix here does follow that convention (`0392579`) but is being rejected. I reproduced locally on this branch and found that on other branches I wasn't seeing this error (nor did I see it on earlier commits here where I was getting flaky e2e errors but not csharp errors). 

My guess is that because the string starts with a 0 it's being interpreted in a way that nuget doesn't like (maybe as a hex number instead of as a string). I've made a tiny change to the README file so that git will produce a different hash for this branch so that the csharp tests will pass.

**Comment by roberthbailey** [comment]
Back to seeing the test flakes in e2e-feature-gates:

```--- FAIL: TestFleetAutoscalerTLSWebhook (312.13s)
    framework.go:284: 
        	Error Trace:	framework.go:284
        	            				fleetautoscaler_test.go:709
        	Error:      	Received unexpected error:
        	            	timed out waiting for the condition
        	Test:       	TestFleetAutoscalerTLSWebhook
        	Messages:   	error waiting for fleet condition on fleet: simple-fleet-dsfck
```

But this means that the csharp sdk code is now happy (for some unknown reason).

File: examples/simple-game-server/README.md [hunk 1]
```diff
@@ -39,13 +39,15 @@ There are some text commands you can send the server to affect its behavior:
...
-| tcp                       | TCP                  | false   |
+| Flag                                   | Environment Variable | Default |
+| -------------------------------------- | -------------------- | ------- |
+| port                                   | PORT                 | 7654    |
+| passthrough                            | PASSTHROUGH          | false   |
+| ready                                  | READY                | true    |
+| automaticShutdownDelaySec              | _n/a_                | 0       |
+| automaticShutdownDelayMin (deprecated) | _n/a_                | 0       |
+| readyDelaySec                          | _n/a_                | 0       |
+| readyIterations                        | _n/a_                | 0       |
```
Comment (ilkercelikyilmaz) [inline examples/simple-game-server/README.md:50]: what happens if the readyDelaySec provided/set but readyIterations not?

**Review by ilkercelikyilmaz — COMMENTED** [review]
LGTM

File: examples/simple-game-server/README.md (hunk 1 above)
Comment (roberthbailey) [reply examples/simple-game-server/README.md:50]: Right now `readyDelaySec` is only used the _first_ time the game server transitions to ready. Each iteration waits `automaticShutdownDelaySec` seconds before going back to ready. 

I wonder if this is too confusing and I should use the `readyDelaySec` between iterations intsead.

**Pull request merged**